### PR TITLE
[#167727303] Setup API documentation using swagger

### PR DESCRIPTION
**Pull request by shonubijerry**
**What does this PR do?**

* Setup swagger documentation using JSON

**Description of Task to be completed?**

* Create a swagger.js document file in /src/docs/swagger
* Create definitions directory and files
* Create swagger paths directory and files 
* Import definitions and paths to swagger.js
* serve swagger.js document file to swagger

**How should this be manually tested?**

* Clone the repository, cd into the repository directory, and run `npm install` finally run `npm run start:dev`
* Go to http://localhost:3000/api-docs to view documentation

**Any background context you want to provide?**

* The project folder structure was added to README.md

**Relevant pivotal tracker stories?**
#167727303

### CODE REVIEW

**Review by joaquinto — CHANGES_REQUESTED** [review]
You did a Great job there however, can you destructure the `errors` from the `errorResponse` , `signIn` and `signinCreate` from the `signInDefinitions` in the `swagger.js` file. Once again great job. 👍👍👍

File: public/api-docs/swagger.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    {
+      name: 'auth',
+      description: 'Everything about authentication'
+    }
+  ],
+  paths: {
+    '/auth/signin': signInPath,
+  },
+  definitions: {
+    signinCreate: signInDefinitions.SigninCreate, // signin request
```
Comment (jsamchineme) [inline public/api-docs/swagger.js]: destructure these definitions from `signInDefinitions`.

**Comment by shonubijerry** [comment]
> You did a Great job there however, can you destructure the `errors` from the `errorResponse` , `signIn` and `signinCreate` from the `signInDefinitions` in the `swagger.js` file. Once again great job.

I think destructuring is not appropriate in this context. After I tried it, the code looks messier and complex. I recommend we don't destructure here.

File: src/index.js [hunk 2]
```diff
@@ -3,7 +3,7 @@ import bodyParser from 'body-parser';
 import logger from 'morgan';
 import cors from 'cors';
 import swaggerUI from 'swagger-ui-express';
-import swaggerDocument from '../public/api-docs/swagger.json';
+import swaggerDocument from '../public/api-docs/swagger';
```
Comment (jsamchineme) [inline src/index.js]: move the `swagger` folder to `src/docs/swagger`

File: README.md [hunk 3]
```diff
@@ -6,4 +6,27 @@
 
 Make company global travel and accommodation easy and convenient for the strong workforce of savvy members of staff, by leveraging the modern web.
 
----
+## Project top-level directory structure
+
+```
+public                          # serve static files used in the swagger documentation
+    ├── api-docs
```
Comment (jsamchineme) [inline README.md]: update the folder structure to reflect the directory change:  `public/api-docs/swagger` --> `src/docs/swagger`

File: src/index.js (hunk 2 above)
Comment (shonubijerry) [reply src/index.js]: Do you mean I should move the `./public/api-docs` contents to `src/docs/swagger`?

**Review by joaquinto — DISMISSED** [review]

**Review by steelze — DISMISSED** [review]

File: src/index.js (hunk 2 above)
Comment (jsamchineme) [reply src/index.js]: yeah.

**Review by joaquinto — DISMISSED** [review]
Nice work man. 👍👍👍

**Review by jsamchineme — APPROVED** [review]

**Review by orley12 — APPROVED** [review]

**Review by SteveFunso — COMMENTED** [review]
So far this is looking ok... and good to be merged

**Pull request merged**